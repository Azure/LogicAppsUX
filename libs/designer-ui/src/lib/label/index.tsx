import { css, Label as FluentLabel } from '@fluentui/react';
import { useIntl } from 'react-intl';

export interface LabelProps {
  className?: string;
  htmlFor?: string;
  id?: string;
  isRequiredField?: boolean;
  text: string;
  tooltip?: string;
}

interface RequiredParameterMarkerProps {
  isRequiredField: boolean;
}

export const Label: React.FC<LabelProps> = ({ className, htmlFor, id, isRequiredField = false, text, tooltip }) => {
  return (
    <FluentLabel className={css(className, 'msla-label')} htmlFor={htmlFor} id={id} title={tooltip || text}>
      <RequiredParameterMarker isRequiredField={isRequiredField} />
      {text}
    </FluentLabel>
  );
};

const RequiredParameterMarker: React.FC<RequiredParameterMarkerProps> = ({ isRequiredField }) => {
  const intl = useIntl();
  if (!isRequiredField) {
    return null;
  }

  const ariaLabel = intl.formatMessage({
    defaultMessage: 'Required',
    description: 'Accessibility prefix for the input label',
  });

  return (
    // eslint-disable-next-line react/jsx-no-literals
    <span className="msla-label-required-parameter" aria-label={ariaLabel}>
      *
    </span>
  );
};
