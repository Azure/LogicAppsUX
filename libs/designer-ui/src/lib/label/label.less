@import (reference) '../variables.less';

.msla-label {
  display: block;
  overflow: hidden;
  text-overflow: ellipsis;
  font-size: @card-label-font-size;
  line-height: 1.25em;
  text-align: left;
  word-wrap: break-word;
}

.msla-label-required-parameter {
  color: #e81123;
  padding-right: 2px;
}

.msla-theme-dark {
  .msla-label-required-parameter {
    color: #f4717c;
  }
}
