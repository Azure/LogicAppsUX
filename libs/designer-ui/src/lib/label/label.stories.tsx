// Button.stories.ts | Button.stories.tsx
import type { LabelProps } from './index';
import { Label } from './index';
import type { ComponentMeta, ComponentStory } from '@storybook/react';
import React from 'react';

export default {
  component: Label,
  title: 'Components/Label',
} as ComponentMeta<typeof Label>;

export const Standard: ComponentStory<typeof Label> = (args: LabelProps) => <Label {...args} />;

Standard.args = {
  text: 'Label',
  isRequiredField: true,
};
