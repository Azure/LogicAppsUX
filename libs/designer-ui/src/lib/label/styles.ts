import { makeStyles, tokens } from '@fluentui/react-components';
import { designTokens } from '../tokens/designTokens';

export const useLabelStyles = makeStyles({
  root: {
    display: 'block',
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    fontSize: designTokens.typography.cardLabelFontSize,
    lineHeight: '1.25em',
    textAlign: 'left',
    wordWrap: 'break-word',
    fontWeight: '600',
  },
  requiredParameterLeft: {
    color: tokens.colorPaletteRedForeground1,
    paddingRight: '2px',
  },
  requiredParameterRight: {
    color: tokens.colorPaletteRedForeground2,
    paddingLeft: '3px',
  },
});
