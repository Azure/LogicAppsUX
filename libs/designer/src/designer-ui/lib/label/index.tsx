import { css, Label as FluentLabel } from '@fluentui/react';
import { useIntl } from 'react-intl';

export enum RequiredMarkerSide {
  LEFT = 'left',
  RIGHT = 'right',
}
export interface LabelProps {
  className?: string;
  htmlFor?: string;
  id?: string;
  isRequiredField?: boolean;
  text: string;
  tooltip?: string;
  requiredMarkerSide?: RequiredMarkerSide;
}

interface RequiredParameterMarkerProps {
  isRequiredField: boolean;
  isRight?: boolean;
}

export const Label: React.FC<LabelProps> = ({
  className,
  htmlFor,
  id,
  isRequiredField = false,
  text,
  tooltip,
  requiredMarkerSide = RequiredMarkerSide.LEFT,
}) => {
  return (
    <FluentLabel className={css(className, 'msla-label')} htmlFor={htmlFor} id={id} title={tooltip || text}>
      {requiredMarkerSide === RequiredMarkerSide.LEFT ? <RequiredParameterMarker isRequiredField={isRequiredField} /> : null}
      {text}
      {requiredMarkerSide === RequiredMarkerSide.RIGHT ? (
        <RequiredParameterMarker isRequiredField={isRequiredField} isRight={true} />
      ) : null}
    </FluentLabel>
  );
};

const RequiredParameterMarker: React.FC<RequiredParameterMarkerProps> = ({ isRequiredField, isRight }) => {
  const intl = useIntl();
  if (!isRequiredField) {
    return null;
  }

  const ariaLabel = intl.formatMessage({
    defaultMessage: 'Required',
    description: 'Accessibility prefix for the input label',
  });

  return (
    // eslint-disable-next-line react/jsx-no-literals
    <span className={isRight ? 'msla-label-required-parameter-right' : 'msla-label-required-parameter'} aria-label={ariaLabel}>
      *
    </span>
  );
};
