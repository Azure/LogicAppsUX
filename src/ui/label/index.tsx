import * as React from 'react';
import { useIntl } from 'react-intl';

export interface LabelProps {
  className?: string;
  htmlFor?: string;
  id?: string;
  isRequiredField?: boolean;
  text: string;
  tooltip?: string;
}

interface RequiredParameterMarkerProps {
  isRequiredField: boolean;
}

const RequiredParameterMarker = (props: RequiredParameterMarkerProps) => {
  const intl = useIntl();
  const ariaLabel = intl.formatMessage({
    defaultMessage: 'Required',
    id: 'qSt0Sb',
    description: 'Accessibility prefix for the input label',
  });
  if (props.isRequiredField) {
    return (
      <span className="msla-label-required-parameter" aria-label={ariaLabel}>
        *
      </span>
    );
  } else {
    return null;
  }
};

export const Label: React.FC<LabelProps> = ({ className, htmlFor, id, isRequiredField, text, tooltip }: LabelProps) => {
  const labelClassName = className ? `${className} msla-label` : 'msla-label';
  const title = tooltip || text;

  return (
    <label className={labelClassName} htmlFor={htmlFor} id={id} title={title}>
      <RequiredParameterMarker isRequiredField={isRequiredField ?? false} />
      {text}
    </label>
  );
};
