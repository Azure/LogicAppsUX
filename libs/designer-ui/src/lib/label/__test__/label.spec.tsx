import renderer from 'react-test-renderer';
import { Label } from '../../label';

describe('lib/label', () => {
  it('should construct', () => {
    const tree = renderer.create(<Label text="label" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('className', () => {
    it('should set the "class" attribute', () => {
      const tree = renderer.create(<Label className="class-name" text="label text" />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('id', () => {
    it('should set the "id" attribute', () => {
      const tree = renderer.create(<Label id="label-id" text="label text" />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('htmlFor', () => {
    it('should set the "for" attribute', () => {
      const tree = renderer.create(<Label htmlFor="an-input" text="label text" />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('isRequiredField', () => {
    it('should render the required parameter marker if set', () => {
      const tree = renderer.create(<Label isRequiredField text="label text" />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('tooltip', () => {
    it('should set the "title" attribute if tooltip is set', () => {
      const tree = renderer.create(<Label isRequiredField text="label text" title="title" />).toJSON();
      const text = 'label text';
    });
  });
});
