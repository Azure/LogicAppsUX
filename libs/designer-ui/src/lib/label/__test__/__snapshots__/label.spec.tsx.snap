// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`lib/label > className > should set the "class" attribute 1`] = `
<label
  className="ms-Label class-name msla-label root-109"
  title="label text"
>
  label text
</label>
`;

exports[`lib/label > htmlFor > should set the "for" attribute 1`] = `
<label
  className="ms-Label msla-label root-109"
  htmlFor="an-input"
  title="label text"
>
  label text
</label>
`;

exports[`lib/label > id > should set the "id" attribute 1`] = `
<label
  className="ms-Label msla-label root-109"
  id="label-id"
  title="label text"
>
  label text
</label>
`;

exports[`lib/label > isRequiredField > should render the required parameter marker if set 1`] = `
<label
  className="ms-Label msla-label root-109"
  title="label text"
>
  <span
    aria-label="Required"
    className="msla-label-required-parameter"
  >
    *
  </span>
  label text
</label>
`;

exports[`lib/label > should construct 1`] = `
<label
  className="ms-Label msla-label root-109"
  title="label"
>
  label
</label>
`;

exports[`lib/label > tooltip > should set the "title" attribute if tooltip is set 1`] = `
<label
  className="ms-Label msla-label root-109"
  title="title"
>
  <span
    aria-label="Required"
    className="msla-label-required-parameter"
  >
    *
  </span>
  label text
</label>
`;
